### PR TITLE
1198 - IdsDataGrid.scrollRowIntoView() does not go the the expected row

### LINKED ISSUE #1198: IdsDatagrid: ScrollRowIntoView does not go the the expected row
**Describe the bug**
When IdsDataGrid.ScrollRowIntoView() does not go the the expected row

**To Reproduce**

1. Open http://localhost:4300/ids-data-grid/virtual-scroll-infinite-scroll.html
2. Scroll down up to the last record (363) and go back to the top
3. On the console log, run document.querySelector("#data-grid-virtual-scroll").scrollRowIntoView(329)
4. The expected record show should be 330 but I am getting row 298

**Expected behavior**
The datagrid needs to show the row 330 

**Version**

- ids enterprise wc 1.0.0-beta.8

**Screenshots**

**Platform**
Infor Application/Team Name: M3 UX
OS Version: Windows 11
Browser Name: [chrome
Browser Version: 109.0.5414.75 (Official Build) (64-bit)

**Additional context**

### PR DESCRIPTION
**Pull request by clayinfor**
**Explain the details for making this change. What existing problem does the pull request solve?**

After triggering an infinite-scroll, `IdsDataGrid.ScrollRowIntoView()` was not going the the expected row.

This PR fixes `IdsDataGrid.ScrollRowIntoView()` so that it finds correct row after infinite-scrolling.

**Related github/jira issue (required)**:

Closes #1198

**Steps necessary to review your pull request (required)**:

1. Open http://localhost:4300/ids-data-grid/virtual-scroll-infinite-scroll.html
2. Scroll down to the last record (300).
3. In the developer console, run:  `document.querySelector("ids-data-grid").scrollRowIntoView(0)`
4. The expected record shown should be 1.
4. In the developer console, run:  `document.querySelector("ids-data-grid").scrollRowIntoView(300)`
5. The expected record shown should be 300.

**Included in this Pull Request**:
- [ ] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

File: doc/CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,8 @@
 ### 1.0.0-beta.9 Fixes
 
 - `[Button]` Renamed "type" attribute to "appearance", mapped a new "type" attribute that sets HTMLButtonElement's "type" attribute. ([#1062](https://github.com/infor-design/enterprise-wc/issues/1062))
+- `[Datagrid]` Fix `IdsDataGrid.ScrollRowIntoView()` so that it finds correct row after infinite-scrolling. ([#1198](https://github.com/infor-design/enterprise-wc/issues/1198))
+=======
```
Comment (tmcconechy) [inline doc/CHANGELOG.md:9]:
```suggestion
```

**Review by tmcconechy — CHANGES_REQUESTED** [review]
@clayinfor i found a bug: for reference the page is 711x806 but i tried a few sizes and the issue seems to not depend on that.

- reload the page (fresh page)
- in the console type `document.querySelector("ids-data-grid").scrollRowIntoView(11)` -> in this case 11 is just out of sight
- then do `document.querySelector("ids-data-grid").scrollRowIntoView(300)` and in this case its only at 58

Whats the rule for where it should scroll to? Should it make it the first row visible? Or just someplace in view? (might need research on scrolIntoView html api)

**Comment by clayinfor** [comment]
Thanks for your feedback @tmcconechy !  The rule is `scrollRowIntoView(x)` basically takes the number you pass in, and multiply it by the configured rowHeight to determine where in the resultset you should be taken to.  If it's not working, then the math was thrown off somewhere.

**Comment by clayinfor** [comment]
hey @tmcconechy, upon further consideration, I think it best to label the latest issue you pointed out above a "won't fix" ... the reason being is because row 300, the one you're trying to scroll to, hasn't been loaded by infinite scroll yet, and so there's no way for our Web Components to know in-advance what the total result-set from the backend will be.

In the case of our `virtual-scroll-infinite-scroll.html` example, we know there are `> 300` rows before we scroll all the way down.  But in a real-world scenario, there's no telling what the last record would be.  And so this is something the end-user should take care of using a `for` or `while` loop etc to keep calling scrollRowIntoView(300) until the infinite-scroll reaches row 300+... but even this would be tricky unless the backend returns a `totalRows` value that the end-user could reference.

**Comment by tmcconechy** [comment]
ok, i wonder if its related to https://github.com/infor-design/enterprise-wc/issues/1199 but we can leave it for now. What about `document.querySelector("ids-data-grid").scrollRowIntoView(11)` where it doesnt fully get it in view?

**Comment by clayinfor** [comment]
Hey @tmcconechy, `scrollRowIntoView(11)` worked fine for me (see screen recording), so... unable to reproduce.

https://user-images.githubusercontent.com/93789065/229194413-0f732d76-487f-41df-81c6-8118eeb2f97f.mov

**Comment by tmcconechy** [comment]
Here is what i see:  https://watch.screencastify.com/v/qvNK2NczY6Su1mgs29DL
So it might depend on screen size....

Maybe it doesn't even need to scroll in this case?

**Comment by clayinfor** [comment]
Oh I see what's going on @tmcconechy ... `scrollRowIntoView(x)` is 0-index based, so when you ran `scrollRowIntoView(11)` in your screen recording, row # 12 is correctly displayed, because index-11 points to row 12... And `scrollRowIntoView(0)` would display row # 1, because row # 1 is `idsDataGrid.data[0]` under the hood.

**Comment by tmcconechy** [comment]
Except in the first case it only went to 8 on the top in the video? So is that correct it just gets it "somewhere" in view? 
Lets see what @GMInfor thinks when testing

**Comment by tmcconechy** [comment]
https://github.com/infor-design/enterprise-wc/issues/1198#issuecomment-1493794516

**Comment by tmcconechy** [comment]
Probably closing this in favor of https://github.com/infor-design/enterprise-wc/pull/1239

**Pull request closed without merging**